src/Razor/Razor.Runtime/src/RazorRazorPageBase.cs

```diff
@@ -0,0 +1,505 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Razor.Runtime.TagHelpers;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Razor
```

Comment:
So to sum up what's here - it looks like a standalone/portable Razor base class would have the following stuff:
- I/O (writing markup and attributes)
- Tag Helpers 
- Layouts (Path/BodyContent/Layout
- Sections 
- Flush

I think we'd also add support for `@inject` - that's pretty essential, and allows you build in a lot of flexibility without putting every feature at the base layer.

Stuff I would cut:
- ViewBag (not something we recommend to users
- URL generation (can't be built-in at this layer)
- "Context" (not clear that this is generalizable)

-----

I'd be pretty happy if this is where we land.